### DISCUSSION TITLE
We want to promote ImperatorLang and are looking for Developer Testimonials!

### DISCUSSION THREAD

**Discussion by CyberTripper666** [OpShin/opshin#65] [Ideas / Improvements]
Hi there! We’re from Teiki, and we want to promote ImperatorLang on our crowdfunding platform, so that people can support Niels in continuing development. We also want to promote general awareness and adoption of ImperatorLang as a PlutusTx alternative.

**So, we're asking for a variety of testimonials that target two audiences:**
- Backers (who may be non-technical)
- Smart Contract developers

**Here are questions we'd like answered:**
- What is one difficulty you had working with PlutusTx? How did ImperatorLang solve that or make it easier for you? You can give us multiple pain points, just make sure you explain for each one what ImperatorLang did for you.
- How easy was it to start working with ImperatorLang? (How much time did it take you to learn or do something? A week to learn? Etc.)
- How has your general experience been working with ImperatorLang? Pretend you're talking to a friend and they're asking how it was and if you recommend it. 
- What are you building or have built with Imperator Lang?
- Do you happen to have any benchmark tests vs Haskell that you can share with us?

Thanks for your time!

**Comment by while0x1** [comment]
> Hi there! We’re from Teiki, and we want to promote ImperatorLang on our crowdfunding platform, so that people can support Niels in continuing development. We also want to promote general awareness and adoption of ImperatorLang as a PlutusTx alternative.
> 
> **So, we're asking for a variety of testimonials that target two audiences:**
> 
> * Backers (who may be non-technical)
> * Smart Contract developers
> 
> **Here are questions we'd like answered:**
> 
> * What is one difficulty you had working with PlutusTx? How did ImperatorLang solve that or make it easier for you? You can give us multiple pain points, just make sure you explain for each one what ImperatorLang did for you.
> * How easy was it to start working with ImperatorLang? (How much time did it take you to learn or do something? A week to learn? Etc.)
> * How has your general experience been working with ImperatorLang? Pretend you're talking to a friend and they're asking how it was and if you if you recommend it.
> * What are you building or have built with Imperator Lang?
> * Do you happen to have any benchmark tests vs Haskell that you can share with us?
> 
> Thanks for your time!

•	What is one difficulty you had working with PlutusTx? How did ImperatorLang solve that or make it easier for you? You can give us multiple pain points, just make sure you explain for each one what ImperatorLang did for you.

_Plutus requires knowledge of Haskell, functional programming and requires a complicated development environment – Nix. Haskell does not use friendly human-readable syntax. The documentation for all the above is ossified and fragmented. If you already have WSL or a python environment eopsin literally installs in seconds. With an attentive development team and growing list of example smart contracts the development experience is comparatively pain-free._   

•	How easy was it to start working with ImperatorLang? (How much time did it take you to learn or do something? A week to learn? Etc.)

_It was the easiest smart contract language to start learning because many makers/developers already have access to an environment with python installed (WSL/Ubuntu). After a few interactions with the ImperatorLang team I had a working contract within a day or two and within a few days had made two valid contracts. ImperatorLang is based on python – one of the most ubiquitous and user-friendly languages ever developed. This means there is an enormous amount of free learning material available to help new developers._ 

•	How has your general experience been working with ImperatorLang? Pretend you're talking to a friend and they're asking how it was and if you if you recommend it.

_ImperatorLang makes smart contract development the easy part of dApp development. I have spent far longer on off-chain code than on implementing smart contracts. I can update, recompile, and use the contracts within transactions in minutes. Because smart contracts are so easy to make, I have opted to use smart contracts to reduce off-chain code. For example, instead of creating a faucet by listening to blockchain events and creating transactions in response, I codified a faucet into an eopsin smart contract which is not only easier than creating an off-chain faucet – it prevents the need to process refunds._ 

•	What are you building or have built with Imperator Lang?

_I built verifair a verifiably fair raffle and NFT distribution platform._
